### DISCUSSION TITLE
How to write  RESTful API?

### DISCUSSION THREAD

**Discussion by FeClickEngineer** [remix-run/remix#2120] [Q&A] [answered]
I did not find the relevant content in the documentation

**Comment by sergiodxa** [comment, marked as answer]
Check https://remix.run/docs/en/v1/guides/api-routes and https://remix.run/docs/en/v1/guides/resource-routes

**Reply by FeClickEngineer** [reply]
So it is here, it is my eyes are clumsy， thank you

**Reply by juniorforlife** [reply]
@sergiodxa seems like `useFetcher` is the only way to call api route or resource route. I get `Uncaught (in promise) TypeError: NetworkError when attempting to fetch resource.` when trying to use fetch instead. Is there any way to fix this?

**Reply by sergiodxa** [reply]
Using fetch directly must work, at the end of the day a resource route is just a normal endpoint.

Are you doing the fetch client-side or server-side? If server-side, remember to include the whole URL (including protocol and host) and not just the pathname.

**Reply by juniorforlife** [reply]
yes I figured out that I must use a full url including the protocol. Wish ‘fetcher.load’ returns a promise like fetch 😔.

**Reply by lopugit** [reply]
@juniorforlife not sure if you found out but you can wrap fetcher in a hook with basically turns it into a promise system

**Reply by tombohub** [reply]
is there openapi generator for those routes?

**Reply by Lrmdom** [reply]
> is there openapi generator for those routes?

Hi. Did you used any generator?

**Reply by tombohub** [reply]
@Lrmdom no, I didn't try to search
